corehq/apps/es/README.rst

```diff
@@ -44,13 +44,21 @@ outages while keeping both up-to-date.
 The least disruptive way to accomplish a multiplexing design is with an adapter
 layer that operates between the low-level third party Elasticsearch Python
 client library and high-level HQ components which need to read/write data in an
-Elasticsearch index. Conveniently, HQ already has the initial framework for this
-layer (the ``ElasticsearchInterface`` class), so the adapter layer is not a new
-concept. The reason the ``ElasticsearchInterface`` implementation cannot
-accommodate multiplexing is because it is not widely adopted for low-level
-Elasticsearch operations (e.g. index creation and configuration), and it does
-not sufficiently decouple low-level Elasticsearch document manipulation logic
-from the high-level HQ code that uses it.
+Elasticsearch index. HQ already has the initial framework for this layer (the
+``ElasticsearchInterface`` class), so the adapter layer is not a new concept.
+The reason that the ``ElasticsearchInterface`` implementation cannot be modified
+in-place to accommodate multiplexing is because it is the wrong level of
+abstraction. The ``ElasticsearchInterface`` abstraction layer was designed as an
+Elasticsearch **version** abstraction. It provides a common set of functions and
+methods so that the high-level HQ "consumer" that uses it can interact with
+Elasticsearch documents without knowing which Elasticsearch *version* is on the
+backend. It is below "index-level" logic, and does not implement index-specific
+functionality needed in order for some indexes to be handled differently than
+others (e.g. some indexes are indexed individually while others are
+multiplexed). The document adapter implementation is a **document** abstraction
```

Comment:
> Is it fair to say _document_ and _index_ are being used interchangeably here?

Hmm, sort of. In the sense that there is only one _Elasticsearch_ "document type" (i.e. mapping) in each index (sidenote: there is only one per index in the HQ implementation, but this will be enforced by Elasticsearch in newer versions). The word "document" here refers to an "Elasticsearch document", which the adapter creates with its `from_python()` method, so it's more than just an index.

> in that we use nested documents which are a level below these abstractions

I'm not sure what you're referring to here.